src/core/stream_recv.c

```diff
@@ -1049,46 +1021,50 @@ QuicStreamReceiveComplete(
     _In_ uint64_t BufferLength
     )
 {
-    if (!Stream->Flags.ReceiveCallPending) {
+    if (Stream->Flags.SentStopSending || Stream->Flags.RemoteCloseFin) {
+        //
+        // The app has aborted their receive path. No need to process any more.
+        //
         return FALSE;
     }
 
-    QuicPerfCounterAdd(QUIC_PERF_COUNTER_APP_RECV_BYTES, BufferLength);
-
-    CXPLAT_FRE_ASSERTMSG(
-        BufferLength <= Stream->RecvPendingLength,
-        "App overflowed read buffer!");
-
-    Stream->Flags.ReceiveCallPending = FALSE;
-
     QuicTraceEvent(
         StreamAppReceiveComplete,
         "[strm][%p] Receive complete [%llu bytes]",
         Stream,
         BufferLength);
 
+    CXPLAT_TEL_ASSERTMSG(
+        BufferLength <= Stream->RecvPendingLength,
+        "App overflowed read buffer!");
+
     //
     // Reclaim any buffer space comsumed by the app.
     //
-    if (Stream->RecvPendingLength == 0 ||
+    if (!Stream->RecvPendingLength ||
```

Comment:
Why changing it to `!Stream->RecvPendingLength`? The implicit conversion from an integer to a boolean makes the code less readable.

Reply:
Maybe `Stream->RecvPendingLength == 0`?

Reply:
+1 on this

Reply:
I changed all the usages to this new model. What do folks prefer, the old version or the new?